javaoop/src/main/java/bootcamp/Car.java

```diff
@@ -43,19 +50,38 @@ public void print() {
     }
 
     // Increments the speed of the car by 1 unit
-    public void goFaster(){
-        this.speed = this.speed + 1;
-        System.out.println("Increased the speed of a car " + this.brand + " " + this.model);
+    public void goFaster() throws Exception {
+        // If clutch pedal is not pressed, then the speed can be manually increased
+        if (!isClutchPressed) {
+            this.speed = this.speed + 1;
+            System.out.println("Increased the speed of a car " + this.brand + " " + this.model);
+        } else {
+            throw new Exception();
+        }
     }
 
     // Decrements the speed of the car by 1 unit
-    public void goSlower(){
-        this.speed = this.speed - 1;
-        System.out.println("Decreased the speed of a car " + this.brand + " " + this.model);
+    public void goSlower() throws Exception {
+        // If clutch pedal is pressed, then speed can be manually decreesed
+        if (isClutchPressed) {
+            this.speed = this.speed - 1;
+            System.out.println("Decreased the speed of a car " + this.brand + " " + this.model);
+        } else {
+            throw new Exception();
+        }
+    }
+
+    // Shifts the gear of a car
+    public void shiftGear(int gearNumber) throws Exception {
+        if (isClutchPressed) {
```

Comment:
add shift to -1 - Reverse speed.
you can only shift to -1 when your speed is 0.